### PR TITLE
fix: remove repetitive code in ListSnapshots

### PR DESCRIPTION
**Pull request by deer-hang**

### CODE REVIEW

File: pkg/evs/controllerserver.go [hunk 1]
```diff
@@ -472,38 +472,9 @@ func (cs *ControllerServer) ListSnapshots(_ context.Context, req *csi.ListSnapsh
...
-		return nil, err
-	}
-	responseEntry := generateListSnapshotsResponseEntry(snapshot)
-	return &csi.ListSnapshotsResponse{Entries: []*csi.ListSnapshotsResponse_Entry{responseEntry}}, nil
-}
-
-func querySnapshotPageList(credentials *config.CloudCredentials, req *csi.ListSnapshotsRequest) (
-	*csi.ListSnapshotsResponse, error) {
 	availableStatus := "available"
 	opts := snapshots.ListOpts{
```
Comment (chengxiangdong) [inline pkg/evs/controllerserver.go:476]: If ID is specified, other parameters need to be removed

File: pkg/evs/controllerserver.go [hunk 2]
```diff
@@ -472,38 +472,9 @@ func (cs *ControllerServer) ListSnapshots(_ context.Context, req *csi.ListSnapsh
...
+		ID:       req.GetSnapshotId(),
 		VolumeID: req.GetSourceVolumeId(),
 		Status:   availableStatus, // Only retrieve snapshots available
 		Limit:    int(req.MaxEntries),
```
Comment (chengxiangdong) [inline pkg/evs/controllerserver.go:480]: `Marker` is missing

File: pkg/evs/controllerserver.go [hunk 3]
```diff
@@ -517,6 +488,7 @@ func querySnapshotPageList(credentials *config.CloudCredentials, req *csi.ListSn
 	for _, element := range pageList.Snapshots {
 		responses = append(responses, generateListSnapshotsResponseEntry(&element))
 	}
+	log.Infof("Successful query snapshot list. detail: %v", protosanitizer.StripSecrets(responses))
 	return &csi.ListSnapshotsResponse{Entries: responses}, nil
```
Comment (chengxiangdong) [inline pkg/evs/controllerserver.go:492]: `NextToken` is missing

File: pkg/evs/controllerserver.go (hunk 1 above)
Comment (deer-hang) [reply pkg/evs/controllerserver.go:476]: OK, adjusted

File: pkg/evs/controllerserver.go (hunk 2 above)
Comment (deer-hang) [reply pkg/evs/controllerserver.go:480]: v2 api does not have marker field

File: pkg/evs/controllerserver.go (hunk 3 above)
Comment (deer-hang) [reply pkg/evs/controllerserver.go:492]: v2 API not match cinder API, v2 API does not have this field

File: pkg/evs/controllerserver.go (hunk 2 above)
Comment (deer-hang) [reply pkg/evs/controllerserver.go]: adjusted

File: pkg/evs/controllerserver.go (hunk 3 above)
Comment (deer-hang) [reply pkg/evs/controllerserver.go]: adjusted

**Pull request merged**